asapdiscovery-ml/asapdiscovery/ml/inference/inference.py

```diff
@@ -211,6 +211,77 @@ def predict_from_smiles(
         data = [self.predict(g) for g in gids]
         data = np.concatenate(np.asarray(data))
         # return a scalar float value if we only have one input
-        if len(data) == 1:
-            data = data[0]
+        if np.all(np.array(data.shape) == 1):
+            data = data.item()
+        return data
+
+
+class StructuralInference(InferenceBase):
+    """
+    Inference class for models that take a structure as input.
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        model_spec: Optional[Path] = None,
+        build_model_kwargs: Optional[dict] = None,
+        device: str = "cpu",
+    ):
+        super().__init__(
+            model_name,
+            self.model_type,
+            model_spec,
+            build_model_kwargs=build_model_kwargs,
+            device=device,
+        )
+
+    def predict(self, pose_dict: dict):
```

Comment:
fairly obvious what this function does, but probably still a good idea to throw a docstring in here